letsencrypt/renewer.py

```diff
@@ -0,0 +1,117 @@
+"""Renewer tool to handle autorenewal and autodeployment of renewed
+certs within lineages of successor certificates, according to
+configuration."""
+
+# TODO: sanity checking consistency, validity, freshness?
+
+# TODO: call new installer API to restart servers after deployment
+
+# TODO: when renewing or deploying, update config file to
+#       memorialize the fact that it happened
+
+import configobj
+import os
+
+from letsencrypt import configuration
+from letsencrypt import client
+from letsencrypt import crypto_util
+from letsencrypt import notify
+from letsencrypt import storage
+from letsencrypt.plugins import disco as plugins_disco
+
+DEFAULTS = configobj.ConfigObj("renewal.conf")
+DEFAULTS["renewal_configs_dir"] = "/tmp/etc/letsencrypt/configs"
+DEFAULTS["official_archive_dir"] = "/tmp/etc/letsencrypt/archive"
+DEFAULTS["live_dir"] = "/tmp/etc/letsencrypt/live"
```

Comment:
First of all, I thinking that the renewal script should accept `--config-dir` and `--work-dir` the same way main CLI script does (moreover, it should default to the same values, i.e. `CLI_DEFAULTS["config_dir"]` and `CLI_DEFAULTS["work_dir"]`) Secondly, it should build the above three paths based on the user-provided values: please see how we handle e.g. `NamespaceConfig.temp_checkpoint_dir` and implement it similarly.